relays/messages-relay/src/message_race_delivery.rs

```diff
@@ -112,10 +116,18 @@ where
 			metrics_msg.update_source_latest_confirmed_nonce::<P>(latest_confirmed_nonce);
 		}
 
+		let new_nonces = if latest_generated_nonce > prev_latest_nonce {
+			self.client
+				.generated_messages_weights(at_block.clone(), prev_latest_nonce + 1..=latest_generated_nonce)
+				.await?
+		} else {
+			BTreeMap::new()
```

Comment:
```suggestion
			MessageWeightsMap::new()
```

Might be more accurate here?